Feature/model constructor

manuGil opened the pull request:
@jgrguric96 @Mano-Rom This is my first attempt to implementing the solution, we draft yesterday. There are many other things to consider when implementing it. Feel free to suggest solution or changes.

jgrguric96 commented on src/illuminator/core/model_constructor.py:
```diff
@@ -0,0 +1,155 @@
+        A flag [input's or output's name] that causes the compute_step method to be called.
+        Only relevant when a model is part of an event-based or hybrid simulator.
+    public: bool
+        ???
```
This is what mosaik defines as public:

> The entry public determines whether a model can be instantiated by a user (True) or if it is a sub-model that cannot be created directly (False)

Mano-Rom commented on src/illuminator/core/model_constructor.py:
```diff
@@ -0,0 +1,155 @@
+        One or more model_parameters, inputs or outputs and their initial value that are of 
+        interest for monitoring or logging during a simulation.
+    triggers: list
+        A flag [input's or output's name] that causes the compute_step method to be called.
```
Yes

manuGil commented:
@Mano-Rom and @jgrguric96 . Although, this PR contains work in progress, I decided to merge it so that we share the same code base. 

@Mano-Rom  I noticed that your version of the `config.yaml` follows one of the initial version of the schema we discussed with Milos. It will be good if we align with a single format/schema before the development move forward. I started with that in this PR. Using your `config.yaml` I updated our `modelling_schema.py` to combine changes you suggested with the ones Milos suggested, and added validation to some of the parts. The `modelling-example.yaml` contains an example of a valid YAML that combines your ideas with Milos ideas. Please take a look and adjust your solutions where needed. You can use the `validate_yaml.py` to validate any of you YAML files.
